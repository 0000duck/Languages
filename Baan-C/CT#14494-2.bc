|
| https://notepad-plus-plus.org/community/topic/14494/functionlist-classrange-question
|

declaration:            |This main section, does not have a sub section.
    func2(10.09, "STR")

before.program:         |This main section, does not have a sub section.
    func1(10)

choice.test.123:        |Has multiple subsections. 2 shown below. Regex to check this would be "choice\s*\.\s*[\w\s*\.\s*]+"
before.choice:          |This sub-section should be listed under choice.test.123:
    func1(20)
after.choice:           |This sub-section should be listed under choice.test.123:
    func2(20.09, "STR")

field.field.123:        |Has multiple subections. 1 shown below. Regex to check this would be "field\s*\.\s*[\w\s*\.\s*]+"
on.input:               |This sub-section should be listed under field.field.123:
    func1(30)

functions:              |Does not have a sub section (per se)... but, encloses a list of functions
                        |which are called from other main/sub sections.
function func1(long var1)   |These are functions, which the existing PR #3702 would work.
{
    | Do Something 1.
}

function string func2(
            double var2,
            string var2_1
            )
{
    | Do Something 2.
}
