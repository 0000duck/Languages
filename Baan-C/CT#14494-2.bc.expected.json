{
    "root":"CT#14494-2.bc"
    "branches":[
        {
            "name":"choice.test.123"
            "leaves":[
                "before.choice",
                "after.choice"
            ]
        },
        {
            "name":"field.field.123"
            "leaves":[
                "on.input"
            ]
        },
        {
            "name":"functions"
            "leaves":[
                "func1",
                "func2"
            ]
        }
    ],
    "leaves":[
        "declaration"
        "before.program"
    ]
}